ICSharpCode.Decompiler/CSharp/Transforms/NormalizeBlockStatements.cs

```diff
@@ -218,6 +218,7 @@ void SimplifyPropertyDeclaration(PropertyDeclaration propertyDeclaration)
 				return;
 			propertyDeclaration.Modifiers |= propertyDeclaration.Getter.Modifiers;
 			propertyDeclaration.ExpressionBody = m.Get<Expression>("expression").Single().Detach();
+			propertyDeclaration.CopyAnnotationsFrom(propertyDeclaration.Getter);
```

Comment:
I left a comment on your old branch where I think these are redundant. After we copy annotations, we end up with a duplicate ILFunction annotation. Have you checked that this works as expected?

Reply:
The property declaration AST node has no ILFunction annotation at all, which makes sense because there is no 1:1 mapping between IL code and metadata, if there are multiple accessors, so we store the ILFunction on the accessors. When removing the accessor syntax from the output, we need to copy all annotations from the single accessor to the other declaration to ensure that the ILFunction is still available to others. This was a small oversight on my end, when I originally implemented that transform.

The problem is that somehow the accessor declaration already has two duplicate ILFunction annotations. This can easily be verified by stepping through CSharpDecompiler.DoDecompile(IProperty, ...) and inspecting the Annotations of the property declaration and accessor declarations step by step.

I will investigate further and hopefully be able to fix the root cause, but in general I don't think that it's a problem having two references to the same ILFunction in the collection. Usually our code uses `FirstOrDefault` semantics when working with annotations.